internal/pkg/template/repository/manager.go

```diff
@@ -17,38 +19,45 @@ const OperationTimeout = 30 * time.Second
 type Manager struct {
 	ctx                 context.Context
 	logger              log.Logger
-	lock                *sync.Mutex
 	defaultRepositories []model.TemplateRepository
+	initGroup           *singleflight.Group
 	repositories        map[string]*git.Repository
 }
 
-func NewManager(ctx context.Context, wg *sync.WaitGroup, logger log.Logger, defaultRepositories []model.TemplateRepository) (*Manager, error) {
+func NewManager(ctx context.Context, serverWg *sync.WaitGroup, logger log.Logger, defaultRepositories []model.TemplateRepository) (*Manager, error) {
 	m := &Manager{
 		ctx:                 ctx,
 		logger:              logger,
-		lock:                &sync.Mutex{},
 		defaultRepositories: defaultRepositories,
+		initGroup:           &singleflight.Group{},
 		repositories:        make(map[string]*git.Repository),
 	}
 
 	// Delete all temp directories on server shutdown
-	wg.Add(1)
+	serverWg.Add(1)
 	go func() {
-		defer wg.Done()
+		defer serverWg.Done()
 		<-ctx.Done()
 		m.Free()
 	}()
 
-	// Add default repositories
+	// Init default repositories in parallel
+	errors := utils.NewMultiError()
+	initWg := &sync.WaitGroup{}
 	for _, repo := range defaultRepositories {
+		repo := repo
 		if repo.Type == model.RepositoryTypeGit {
-			if err := m.AddRepository(repo); err != nil {
-				return nil, err
-			}
+			initWg.Add(1)
+			go func() {
+				defer initWg.Done()
+				if _, err := m.Repository(repo); err != nil {
+					errors.Append(err)
+				}
+			}()
```

Comment:
Namiesto metody `AddRepository` je metoda `Repository`, kedze robili vlastne to iste, len jedna vracala iba chybu a druha aj hodnotu.

`Repository` sa moze spustit uz paralelne s roznymi repositories, uz tam nie je `lock`.
(Teda spustit sa mohla aj predtym, ale inicializacia by nebola paralelna).